test/filt_stream.jl

```diff
@@ -312,16 +316,15 @@ function test_arbitrary(Th, x, resampleRate, numFilters)
     reset!(myfilt)
     piecewiseResult = similar(x, 0)
     sizehint!(piecewiseResult, ceil(Int, length(x)*resampleRate))
-    @timeifinteractive for i in 1:length(x)
-        thisY = filt(myfilt, x[i:i])
+    one_buffer = similar(x, 1)
+    @timeifinteractive for p in x
+        thisY = filt(myfilt, setindex!(one_buffer, p, 1))
         append!(piecewiseResult, thisY)
     end
 
-    commonLen = minimum(length.((naiveResult, statelessResult, statefulResult, piecewiseResult)))
-    resize!(naiveResult, commonLen)
-    resize!(statelessResult, commonLen)
-    resize!(statefulResult, commonLen)
-    resize!(piecewiseResult, commonLen)
+    results = (naiveResult, statelessResult, statefulResult, piecewiseResult)
+    commonLen = minimum(length, results)
+    resize!.(results, commonLen)
```

Comment:
I'd prefer the more explicit form here. This line changing all of the `xxxResult`s is a bit non-obvious. (Using `minimum(length, ...)` instead of `minimum(length.(...))` is good, though.)